[バグ] lintのエラー修正(`/app/src/components/ImageEditor.tsx`)

## バグの概要
`make lint`コマンドで警告が発生しているので解消する。

## 期待される動作

警告・エラーが発生しない。

## 実際の動作

警告・エラーが発生する。

## その他の情報

現状の`make lint`の結果は以下の通りです。

```bash
/app/src/components/ImageEditor.tsx
  141:6  warning  React Hook useEffect has a missing dependency: 'imgSrc'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

--- a/src/components/ImageEditor.tsx
+++ b/src/components/ImageEditor.tsx
@@ -114,22 +114,29 @@ export default function ImageEditor({
     };
   }, [imgSrc]);
 
-  // ファイルからURLを生成（imgSrcに依存しない形にリファクタ）
+  // ファイルからURLを生成（imgSrcに依存しない形 + FileReaderのクリーンアップ追加）
   useEffect(() => {
-    if (!imageFile) return;
+    if (!imageFile) {
+      setImgSrc('');
+      return;
+    }
 
     // 一旦クリアしてから少し待って新しい画像を設定
     setImgSrc('');
+
+    const reader = new FileReader();
+    const handleLoad = () => {
+      const result = reader.result?.toString() || '';
+      setImgSrc(result);
+    };
+    reader.addEventListener('load', handleLoad);
+
     const timerId = window.setTimeout(() => {
-      const reader = new FileReader();
-      reader.addEventListener('load', () => {
-        const result = reader.result?.toString() || '';
-        setImgSrc(result);
-      });
       reader.readAsDataURL(imageFile);
     }, 100);
 
     return () => {
+      reader.removeEventListener('load', handleLoad);
       window.clearTimeout(timerId);
     };
   }, [imageFile]);

--- a/src/components/ImageEditor.tsx
+++ b/src/components/ImageEditor.tsx
@@ -99,20 +99,15 @@ export default function ImageEditor({
   const [isImageLoaded, setIsImageLoaded] = useState(false);
   const [isSaving, setIsSaving] = useState(false);
 
-  // コンポーネントのクリーンアップ
+  // コンポーネントのクリーンアップ（Hammer.jsの解放のみ）
   useEffect(() => {
     return () => {
-      // URLを解放
-      if (imgSrc.startsWith('blob:')) {
-        URL.revokeObjectURL(imgSrc);
-      }
-      // Hammer.jsのインスタンスを解放
       if (hammerInstanceRef.current) {
         hammerInstanceRef.current.destroy();
         hammerInstanceRef.current = null;
       }
     };
-  }, [imgSrc]);
+  }, []);
 
   // ファイルからURLを生成（URL.createObjectURL + requestAnimationFrame）
   useEffect(() => {

--- a/src/components/ImageEditor.tsx
+++ b/src/components/ImageEditor.tsx
@@ -114,30 +114,24 @@ export default function ImageEditor({
     };
   }, [imgSrc]);
 
-  // ファイルからURLを生成（imgSrcに依存しない形 + FileReaderのクリーンアップ追加）
+  // ファイルからURLを生成（URL.createObjectURL + requestAnimationFrame）
   useEffect(() => {
     if (!imageFile) {
       setImgSrc('');
       return;
     }
 
-    // 一旦クリアしてから少し待って新しい画像を設定
+    // 一旦クリアしてから、次フレームで新しいURLを設定
     setImgSrc('');
 
-    const reader = new FileReader();
-    const handleLoad = () => {
-      const result = reader.result?.toString() || '';
-      setImgSrc(result);
-    };
-    reader.addEventListener('load', handleLoad);
-
-    const timerId = window.setTimeout(() => {
-      reader.readAsDataURL(imageFile);
-    }, 100);
+    const objectUrl = URL.createObjectURL(imageFile);
+    const animationFrameId = window.requestAnimationFrame(() => {
+      setImgSrc(objectUrl);
+    });
 
     return () => {
-      reader.removeEventListener('load', handleLoad);
-      window.clearTimeout(timerId);
+      window.cancelAnimationFrame(animationFrameId);
+      URL.revokeObjectURL(objectUrl);
     };
   }, [imageFile]);
 

--- a/src/components/ImageEditor.tsx
+++ b/src/components/ImageEditor.tsx
@@ -114,30 +114,24 @@ export default function ImageEditor({
     };
   }, [imgSrc]);
 
-  // ファイルからURLを生成
+  // ファイルからURLを生成（imgSrcに依存しない形にリファクタ）
   useEffect(() => {
-    if (imageFile) {
-      // 既存のURLを解放
-      if (imgSrc) {
-        setImgSrc('');
-        // 少し待ってから新しい画像を設定
-        setTimeout(() => {
-          const reader = new FileReader();
-          reader.addEventListener('load', () => {
-            const result = reader.result?.toString() || '';
-            setImgSrc(result);
-          });
-          reader.readAsDataURL(imageFile);
-        }, 100);
-      } else {
-        const reader = new FileReader();
-        reader.addEventListener('load', () => {
-          const result = reader.result?.toString() || '';
-          setImgSrc(result);
-        });
-        reader.readAsDataURL(imageFile);
-      }
-    }
+    if (!imageFile) return;
+
+    // 一旦クリアしてから少し待って新しい画像を設定
+    setImgSrc('');
+    const timerId = window.setTimeout(() => {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        const result = reader.result?.toString() || '';
+        setImgSrc(result);
+      });
+      reader.readAsDataURL(imageFile);
+    }, 100);
+
+    return () => {
+      window.clearTimeout(timerId);
+    };
   }, [imageFile]);
 
   // 画像がロードされたときの処理

--- a/src/components/user/UserSettingsModal.tsx
+++ b/src/components/user/UserSettingsModal.tsx
@@ -141,7 +141,9 @@ export default function UserSettingsModal({ isOpen, onClose, profile }: UserSett
           });
           break;
       }
-    } catch (_error) {
+    } catch (error) {
+      // 詳細は開発者向けにログ出力し、ユーザーには汎用メッセージを表示
+      console.error('更新処理に失敗しました:', error);
       alert('更新に失敗しました');
     }
   };

--- a/src/components/user/UserSettingsModal.tsx
+++ b/src/components/user/UserSettingsModal.tsx
@@ -141,8 +141,8 @@ export default function UserSettingsModal({ isOpen, onClose, profile }: UserSett
           });
           break;
       }
-    } catch (error) {
-      alert(error instanceof Error ? error.message : '更新に失敗しました');
+    } catch (_error) {
+      alert('更新に失敗しました');
     }
   };
 

--- a/src/test/setup.ts
+++ b/src/test/setup.ts
@@ -48,20 +48,32 @@ afterEach(() => {
 // afterAll(() => server.close());
 
 // Supabaseのモック
-vi.mock('@supabase/supabase-js', () => ({
-  createClient: vi.fn(() => ({
-    auth: {
-      signInWithOAuth: vi.fn(),
-      signOut: vi.fn(),
-      onAuthStateChange: vi.fn(),
-      getSession: vi.fn(),
-    },
-    from: vi.fn(),
-    storage: {
+vi.mock('@supabase/supabase-js', () => {
+  class AuthApiError extends Error {
+    status?: number;
+    constructor(message: string, status?: number) {
+      super(message);
+      this.name = 'AuthApiError';
+      this.status = status;
+    }
+  }
+
+  return {
+    AuthApiError,
+    createClient: vi.fn(() => ({
+      auth: {
+        signInWithOAuth: vi.fn(),
+        signOut: vi.fn(),
+        onAuthStateChange: vi.fn(),
+        getSession: vi.fn(),
+      },
       from: vi.fn(),
-    },
-  })),
-}));
+      storage: {
+        from: vi.fn(),
+      },
+    })),
+  };
+});
 
 // react-helmet-asyncのモック
 vi.mock('react-helmet-async', () => ({

